website/docs/r/transfer_server.html.markdown

```diff
@@ -0,0 +1,89 @@
+---
+layout: "aws"
+page_title: "AWS: aws_transfer_server"
+sidebar_current: "docs-aws-resource-transfer-server"
+description: |-
+  Provides a AWS Transfer Server resource.
+---
+
+# aws_transfer_server
+
+Provides a AWS Transfer Server resource.
+
+
+```hcl
+resource "aws_iam_role" "foo" {
+	name = "tf-test-transfer-server-iam-role"
+  
+	assume_role_policy = <<EOF
+{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+		"Effect": "Allow",
+		"Principal": {
+			"Service": "transfer.amazonaws.com"
+		},
+		"Action": "sts:AssumeRole"
+		}
+	]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "foo" {
+	name = "tf-test-transfer-server-iam-policy-%s"
+	role = "${aws_iam_role.foo.id}"
+	policy = <<POLICY
+{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+		"Sid": "AllowFullAccesstoCloudWatchLogs",
+		"Effect": "Allow",
+		"Action": [
+			"logs:*"
+		],
+		"Resource": "*"
+		}
+	]
+}
+POLICY
+}
+
+
+resource "aws_transfer_server" "foo" {
+  identity_provider_type = "SERVICE_MANAGED"
+  logging_role = "${aws_iam_role.foo.arn}"
+
+  tags {
+	NAME   = "tf-acc-test-transfer-server"
+	ENV    = "test"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `invocation_role` - (Optional) Amazon Resource Name (ARN) of the IAM role used to authenticate the user account with an `identity_provider_type` of `API_GATEWAY`.
+* `url` - (Optional) - URL of the service endpoint used to authenticate users with an `identity_provider_type` of `API_GATEWAY`.
+* `identity_provider_type` - (Optional) The mode of authentication enabled for this service. The default value is `SERVICE_MANAGED`, which allows you to store and access SFTP user credentials within the service. `API_GATEWAY` indicates that user authentication requires a call to an API Gateway endpoint URL provided by you to integrate an identity provider of your choice.
+* `logging_role` - (Optional) Amazon Resource Name (ARN) of an IAM role that allows the service to write your SFTP users’ activity to your Amazon CloudWatch logs for monitoring and auditing purposes.
+* `tags` - (Optional) A mapping of tags to assign to the resource.
```

Comment:
these docs say that the tags are optional, but applying a resource without tags doesn't work:

```
Error: Error applying plan:

1 error(s) occurred:

* aws_transfer_server.sftp_server: 1 error(s) occurred:

* aws_transfer_server.sftp_server: Error creating Transfer Server: InvalidParameter: 1 validation error(s) found.
- minimum field size of 1, CreateServerInput.Tags.
```

i'm happy to open a PR to fix, but i guess i wanna verify that these are meant to be optional. i know that the AWS API doesn't require the tags, but i don't know if terraform wants them to be required?